docs/_snippets/examples/custom.css

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+:root {
+	/* Overrides the border-radius setting in the theme. */
+	--ck-border-radius: 4px;
+
+	/* Overrides the default font size in the theme. */
+	--ck-font-size-base: 14px;
+
+	/* Helper variables to avoid duplication in the colors. */
+	--ck-custom-background: #4A494B;
+	--ck-custom-foreground: #3a393d;
+	--ck-custom-border: #383738;
+	--ck-custom-white: #fff;
+
+	/* -- Overrides generic colors -------------------------------------------------------------- */
+
+	--ck-color-focus-border: #48a3f5;
+	--ck-color-text: #fafafa;
+	--ck-color-shadow-drop: rgba( 0, 0, 0, .2 );
+	--ck-color-shadow-inner: rgba( 0, 0, 0, .1 );
+
+	/* -- Overrides the default .ck-button class colors ----------------------------------------- */
+
+	--ck-color-button-default-background: var(--ck-custom-background);
+	--ck-color-button-default-border: var(--ck-custom-border);
+	--ck-color-button-default-focus-background: #434244;
+	--ck-color-button-default-focus-border: #323232;
+	--ck-color-button-default-active-background: #3f3e40;
+	--ck-color-button-default-active-border: #302f30;
+	--ck-color-button-default-active-shadow: #3b3a3c;
+	--ck-color-button-default-disabled-background: var(--ck-custom-background);
+	--ck-color-button-default-disabled-border: var(--ck-custom-border);
+
+	--ck-color-button-on-background: var(--ck-custom-foreground);
+	--ck-color-button-on-border: var(--ck-custom-border);
+	--ck-color-button-on-focus-background: #343337;
+	--ck-color-button-on-focus-border: #323232;
+	--ck-color-button-on-active-background: #313034;
+	--ck-color-button-on-active-border: #302f30;
+	--ck-color-button-on-active-shadow: #2e2e31;
+	--ck-color-button-on-disabled-background: var(--ck-custom-foreground);
+	--ck-color-button-on-disabled-border: var(--ck-custom-border);
+
+	--ck-color-button-action-background: #1ABC9C;
+	--ck-color-button-action-border: #49d2b7;
+	--ck-color-button-action-focus-background: #17a98c;
```

Comment:
As we talked we can't use postcss plugin for colors with functions like darken/lighten so maybe should we use `hsl/hsla` format? It be easier to understand & modify shades of color.

```css
--ck-color-button-action-background: hsl( 168, 76%, 42% )
--ck-color-button-action-focus-background: hsl( 168, 76%, 38% )
```

Reply:
That's a great idea. And what's more, you can use variables and `calc` to [shift the color](https://jsfiddle.net/oleq/5rwguy1L/).

```css
:root {
  --lightness-shift: 30%;
}

#first {
  background: hsl(120,100%, 50%);
}

#second {
  background: hsl(120,100%, calc(50% - var(--lightness-shift)));
}
```

Reply:
...which may be broken in Firefox (and IE11?). Needs research.

Reply:
Perfect! I forgot about changing the shade of colour by a variable in HSL.

[HSL is full supported since IE8+ so no worries here.](https://caniuse.com/#search=hsl)

Reply:
[Pity we won't be able to use `calc()` with `var()`](https://codepen.io/thebabydino/pen/wfraH). I checked that out, it does not work in FF and IE11.

Reply:
Still, `hsl` sounds like a plan.